types/slashCommandConfig.ts

```diff
@@ -0,0 +1,9 @@
+import { RESTPostAPIApplicationCommandsJSONBody } from 'discord-api-types';
+import { CommandInteraction } from 'discord.js';
+
+export interface ISlashCommandConfig {
```

Comment:
With the risk of sounding like a "tabs vs spaces" kind of person, I personally dislike having the `I` added to the name of the interface. If I need to know the type, I can hover over the element in my IDE. For me at least, it's the equivalent of naming a function something like `funcAdd`.

Reply:
This is a good nit! At some point, the TypeScript team [took an official stance against it](https://github.com/microsoft/TypeScript-Handbook/issues/121) in the handbook.

Reply:
Thanks for the link, that was a good read. I am more than happy to drop that prefix, I originally wrote it without one, but added it after doing going through a more mature TypeScript codebase and seeing that used everywhere.

Reply:
The latest commit removes all of the I prefixes from interfaces.